src/matlab/model_reformulation_nosnoc.m

```diff
@@ -36,7 +36,9 @@
 end
 
 unfold_struct(model,'caller');
+settings_bkp = settings;
 unfold_struct(settings,'caller')
+settings = settings_bkp;
```

Comment:
Why is that needed?

Reply:
Because for a reason I still have not hunted down, `unfold_struct` sometimes clobbers some members of its argument in the caller scope.  I will drop a comment on these.

Reply:
ok, sorry I actually saw a comment at some point. It indeed seems very weird behavior though :sweat_smile: